File: VB/MyLayoutView/MyLayoutViewInfoRegistrator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.Handler
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace MyXtraGrid
	Public Class MyLayoutViewInfoRegistrator
		Inherits LayoutViewInfoRegistrator
		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyLayoutView"
			End Get
		End Property
		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyLayoutView(TryCast(grid, GridControl))
		End Function
		Public Overrides Function CreateViewInfo(ByVal view As BaseView) As BaseViewInfo
			Return New MyLayoutViewInfo(TryCast(view, MyLayoutView))
		End Function
		Public Overrides Function CreateHandler(ByVal view As BaseView) As BaseViewHandler
			Return New MyLayoutViewHandler(TryCast(view, MyLayoutView))
		End Function
	End Class
End Namespace

File: CS/MyLayoutView/MyLayoutViewInfoRegistrator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.Handler;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace MyXtraGrid
{
    public class MyLayoutViewInfoRegistrator : LayoutViewInfoRegistrator
    {
        public override string ViewName { get { return "MyLayoutView"; } }
        public override BaseView CreateView(GridControl grid) { return new MyLayoutView(grid as GridControl); }
        public override BaseViewInfo CreateViewInfo(BaseView view) { return new MyLayoutViewInfo(view as MyLayoutView); }
        public override BaseViewHandler CreateHandler(BaseView view) { return new MyLayoutViewHandler(view as MyLayoutView); }
    }
}
